src/Compilers/Core/Portable/MetadataReader/PEModule.cs

```diff
@@ -1903,18 +1904,21 @@ internal static bool CrackStringArrayInAttributeValue(out ImmutableArray<string?
             if (sig.RemainingBytes >= 4)
             {
                 uint arrayLen = sig.ReadUInt32();
-                var stringArray = new string?[arrayLen];
-                for (int i = 0; i < arrayLen; i++)
+                if (sig.RemainingBytes >= arrayLen)
```

Comment:
>sig.RemainingBytes >= arrayLen [](http://example.com/codeflow?start=20&length=30)

Since the 0xFFFFFFFF value is part of the specification we probably should have an explicit equality check for it. Same in other similar functions. Also, while in ```CrackByteArrayInAttributeValue``` and ```CrackBoolArrayInAttributeValue``` it is appropriate to compare amount of  bytes to the amount of elements, I don not think it is appropriate here. Even if we have that many bytes, it doesn't mean we have that many strings.